crates/astria-core/src/sequencer/v1alpha1/block/mod.rs

```diff
@@ -1275,3 +1291,71 @@ enum DepositErrorKind {
     #[error("the asset ID length is not 32 bytes")]
     IncorrectAssetIdLength(#[source] asset::IncorrectAssetIdLength),
 }
+
+/// A piece of data that is sent to a rollup execution node.
+///
+/// The data can be either sequenced data (originating from a [`SequenceAction`]
+/// submitted by a user) or a [`Deposit`] originating from a [`BridgeLockAction`].
+///
+/// The rollup node receives this type, protobuf-encoded, from conductor and
+/// must decode it accordingly.
+#[derive(Debug, Clone, PartialEq)]
+pub enum RollupData {
+    SequencedData(Vec<u8>),
+    Deposit(Deposit),
+}
+
+impl RollupData {
+    #[must_use]
+    pub fn into_raw(self) -> raw::RollupData {
+        match self {
+            Self::SequencedData(data) => raw::RollupData {
+                value: Some(raw::rollup_data::Value::SequencedData(data)),
+            },
+            Self::Deposit(deposit) => raw::RollupData {
+                value: Some(raw::rollup_data::Value::Deposit(deposit.into_raw())),
+            },
+        }
+    }
+
+    /// Attempts to transform the `RollupData` from its raw representation.
+    ///
+    /// # Errors
+    ///
+    /// - if the `data` field is not set
+    /// - if the variant is `Deposit` but a `Deposit` cannot be constructed from the raw proto
+    pub fn try_from_raw(raw: raw::RollupData) -> Result<Self, RollupDataError> {
+        let raw::RollupData {
+            value,
+        } = raw;
+        match value {
+            Some(raw::rollup_data::Value::SequencedData(data)) => Ok(Self::SequencedData(data)),
+            Some(raw::rollup_data::Value::Deposit(deposit)) => Deposit::try_from_raw(deposit)
+                .map(Self::Deposit)
+                .map_err(RollupDataError::deposit),
+            None => Err(RollupDataError::field_not_set("data")),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct RollupDataError(RollupDataErrorKind);
+
+impl RollupDataError {
+    fn field_not_set(field: &'static str) -> Self {
+        Self(RollupDataErrorKind::FieldNotSet(field))
+    }
+
+    fn deposit(source: DepositError) -> Self {
+        Self(RollupDataErrorKind::Deposit(source))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum RollupDataErrorKind {
+    #[error("the expected field in the raw source type was not set: `{0}`")]
+    FieldNotSet(&'static str),
+    #[error(transparent)]
```

Comment:
Please don't make this `transparent`, but give it a proper description, in this case probably:
```suggestion
    #[error("failed to validate `deposit` field)]
```